### PR TITLE
Implement Page.SelectAsync

### LINKED ISSUE #51: Implement Page.Select tests
https://github.com/GoogleChrome/puppeteer/blob/v1.0.0/test/test.js#L3044

### PR DESCRIPTION
**Pull request by kblok**
closes #51

### CODE REVIEW

**Comment by kblok** [comment]
Ready to review @Meir017

**Comment by Meir017** [comment]
looks like you're missing the test
* `should deselect all options when passed no values for a select without multiple`

File: lib/PuppeteerSharp/Frame.cs [hunk 1]
```diff
@@ -315,6 +316,22 @@ internal Task<JSHandle> WaitForFunctionAsync(string script, WaitForFunctionOptio
             return handle.AsElement();
         }
 
+        internal async Task<string[]> SelectAsync(string selector, object[] args)
```
Comment (Meir017) [inline lib/PuppeteerSharp/Frame.cs]: make `params`?

File: lib/PuppeteerSharp/Frame.cs [hunk 2]
```diff
@@ -315,6 +316,22 @@ internal Task<JSHandle> WaitForFunctionAsync(string script, WaitForFunctionOptio
...
+        {
+            return await QuerySelectorAsync(selector).EvaluateFunctionAsync<string[]>(@"(element, values) => {
```
Comment (Meir017) [inline lib/PuppeteerSharp/Frame.cs]: arrow function? (not completely sure about this one)

File: lib/PuppeteerSharp/Frame.cs (hunk 1 above)
Comment (Meir017) [inline lib/PuppeteerSharp/Frame.cs]: redundant `async` and `await` keywords. just return ` QuerySelectorAsync(...`

Comment (Meir017) [inline lib/PuppeteerSharp/Frame.cs]: in puppeteer the `args` is called `values` and if of type `string[]` not `object[]`

Comment (kblok) [reply lib/PuppeteerSharp/Frame.cs]: I didn't know that `params string[]` was valid, that's cool.

Comment (Meir017) [reply lib/PuppeteerSharp/Frame.cs]: params works with `T[]` 😄

**Comment by kblok** [comment]
Done @Meir017

**Pull request merged**